index.html

```diff
@@ -90,20 +91,33 @@ <h1>APIs</h1>
         el: "#app",
         data: {
             username: logic.initialise(),
-            loginError: ""
+            loginError: "",
+            redirectMessage: ""
         },
         methods: {
             logout: function() {
                 const data = this;
                 logic.logout().then(
-                    () => { data.username = ''; data.loginError = '' },
+                    () => {
+                        data.username = '';
+                        data.loginError = '';
+                    },
                     (error) => { data.username = ''; data.loginError = error }
                 );
             },
             login: function(email, password) {
                 const data = this;
                 logic.login(email, password).then(
-                    (result) => { data.username = result; data.loginError = ''; },
+                    (result) => {
+                        data.username = result;
+                        data.loginError = '';
+                        const redirectLocation = MontaguLogin.redirectLocation(location.search);
+                        if (redirectLocation) {
+                            data.redirectMessage = `Redirecting you back to ${redirectLocation}.
+                            If you are not automatically redirected in a few seconds please click <a href="${redirectLocation}">here</a>`;
+                            window.location.href=redirectLocation
```

Comment:
I'm not doing any validation on this parameter in the end as I couldn't see any reason to?